spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/support/MongoRepositoryFactory.java

```diff
@@ -34,18 +34,23 @@
 import org.springframework.data.repository.core.NamedQueries;
 import org.springframework.data.repository.core.RepositoryMetadata;
 import org.springframework.data.repository.core.support.RepositoryFactorySupport;
+import org.springframework.data.repository.query.EvaluationContextProvider;
 import org.springframework.data.repository.query.QueryLookupStrategy;
 import org.springframework.data.repository.query.QueryLookupStrategy.Key;
 import org.springframework.data.repository.query.RepositoryQuery;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
 import org.springframework.util.Assert;
 
 /**
  * Factory to create {@link MongoRepository} instances.
  * 
  * @author Oliver Gierke
+ * @author Thomas Darimont
  */
 public class MongoRepositoryFactory extends RepositoryFactorySupport {
 
+	private static final SpelExpressionParser EXPRESSION_PARSER = new SpelExpressionParser();
```

Comment:
We should move this `SpelExpressionParser` to RepositoryFactorySupport - this could then be shared between multiple store modules.